libraries/AP_RangeFinder/RangeFinder_Backend.h

```diff
@@ -80,8 +79,12 @@ class AP_RangeFinder_Backend
     void set_status(RangeFinder::RangeFinder_Status status);
 
     RangeFinder::RangeFinder_State &state;
-    MAV_DISTANCE_SENSOR sensor_type;
 
     // semaphore for access to shared frontend data
     AP_HAL::Semaphore *_sem;    
+
+    virtual MAV_DISTANCE_SENSOR _get_mav_distance_sensor_type() const {
+        // most of the backends are lasers:
+        return MAV_DISTANCE_SENSOR_LASER;
```

Comment:
I disagree with this. It is true that most are lasers, but I don't agree this is a good way to do the API - it should return UNKNOWN unless the driver itself says what type it is.